docs/definition.rst

```diff
@@ -0,0 +1,185 @@
+=====================
+Definer/Definition Trait
+=====================
+
+.. php:trait:: DefinerTrait
```

Comment:
from the documentation here it's not obvious what the benefit is.

My understanding is vague. 

I think it allows you to use a 3rd party app class without extending it or writing any code. Injecting it through config files, so you have to define this pattern.

It's not usable for ATK itself and from the initial doc it looks like it does same thing as factory but more complex and unorthodox. Try looking at it as someone who is not familiar to Pimple 🤯

Reply:
I just adapted something that i already use and is really good to have all configurations or classes in a folder distributed in multiple files, that can be called with only one line of code only when needed, if first argument is FQCN the editor probably can do something for intellisense. Obviously for a small project can be only something in plus to be defined, but for medium to big project i think having configurations in a single place is a must with this can be done with lazy object instantiation. i'm sorry if in docs i was not so good to pass the concept and trust me is not so unorthodox PHP-DI and Pimple are used everywhere in php community.

Reply:
The implementation itself is good, but I still struggle to see why not simply do:

```
$app->logger = include 'config/logger.php';
```

It's not like we'll want multiple loggers in our app. Is there any other example? I'd like to hear from others that:
 - yes, we can't live without it!
 - yes, it must be in ATK core, can't be elsewhere!

Reply:
@romaninsh I'm not sure too, just figure out if it can be useful, i see more and more glitch trying to be compliant with PSR-11 ContainerInterface, just too many repetition in code.

For me can be closed.

Reply:
@abbadon1334 lets still discuss it. There is some possibility it can help us initialize $app->logger and $app->translator and other things like that.

It's not a very good practice to rely in atk/core on non-existing property of non-existing object.